misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -373,15 +412,196 @@ private void createReport(final IBundleCoverage bundleCoverage, Map<String, Modu
                                     coveredLines.add(i);
                                 }
                             }
-                            Document document = getDocument(sourceFileCoverage.getName());
                             if (document != null) {
                                 moduleCoverage.addSourceFileCoverage(document, coveredLines,
                                         missedLines);
                             }
                             moduleCoverageMap.put(sourceFileModule, moduleCoverage);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void filterGeneratedCoverage(Map<String, ModuleCoverage> moduleCoverageMap, Module originalModule) {
+
+        for (DocumentId documentId : originalModule.documentIds()) {
+            Document originalDocument = originalModule.document(documentId);
+            Document document = getDocument(originalModule.moduleName().toString(), originalDocument.name());
+
+            try {
+                // Use diff utils to analyze the text lines in the doc
+                Patch<String> stringPatch = DiffUtils.diff(originalDocument.textDocument().textLines(),
+                        document.textDocument().textLines());
+
+                if (!stringPatch.getDeltas().isEmpty()) {
+                    List<AbstractDelta<String>> patchDeltas = stringPatch.getDeltas();
+                    List<Integer> modifiedLines = new ArrayList();
```

Comment:
We are capturing the insertions here. Let's rename the variable to reflect that. E.g.: insertions / insertedLines